jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/client/ClientUpgradeRequest.java

```diff
@@ -224,19 +230,17 @@ public void onComplete(Result result)
             }
 
             Throwable failure = result.getFailure();
-            if ((failure instanceof java.net.SocketException) ||
+            if (!((failure instanceof java.net.SocketException) ||
                 (failure instanceof java.io.InterruptedIOException) ||
                 (failure instanceof HttpResponseException) ||
-                (failure instanceof UpgradeException))
-            {
-                // handle as-is
-                handleException(failure);
-            }
-            else
+                (failure instanceof UpgradeException)))
             {
                 // wrap in UpgradeException
-                handleException(new UpgradeException(requestURI, responseStatusCode, responseLine, failure));
+                failure = new UpgradeException(requestURI, responseStatusCode, responseLine, failure);
             }
+
+            handleException(failure);
+            frameHandler.onError(failure, Callback.NOOP);
```

Comment:
IMO it's more readable as it was before :)

Reply:
I would call `frameHandler.onError()` from within `handleException()` so that we're sure it's called in all failure cases.